src/main/java/com/redhat/rhjmc/containerjfr/net/SslConfiguration.java

```diff
@@ -0,0 +1,207 @@
+package com.redhat.rhjmc.containerjfr.net;
+
+import com.redhat.rhjmc.containerjfr.core.log.Logger;
+import com.redhat.rhjmc.containerjfr.core.sys.Environment;
+import com.redhat.rhjmc.containerjfr.core.sys.FileSystem;
+import io.vertx.core.http.HttpServerOptions;
+import io.vertx.core.net.JksOptions;
+import io.vertx.core.net.PemKeyCertOptions;
+import io.vertx.core.net.PfxOptions;
+import org.openjdk.jmc.common.util.Pair;
+
+import java.nio.file.Path;
+
+class SslConfiguration {
+    private final Environment env;
+    private final FileSystem fs;
+    private final Logger logger;
+
+    private final Path keystorePath;
+    private final String keystorePass;
+
+    private final Path keyPath;
+    private final Path certPath;
+
+    private static final String KEYSTORE_PATH_ENV = "KEYSTORE_PATH";
+    private static final String KEYSTORE_PASS_ENV = "KEYSTORE_PASS";
+    private static final String KEY_PATH_ENV = "KEY_PATH";
+    private static final String CERT_PATH_ENV = "CERT_PATH";
+
+    SslConfiguration(Environment env, FileSystem fs, Logger logger) throws IllegalArgumentException {
+        this.env = env;
+        this.fs = fs;
+        this.logger = logger;
+
+        {
+            Path path = obtainKeyStorePathIfSpecified();
+            if (path != null) {
+                keystorePath = path;
+                keystorePass = env.getEnv(KEYSTORE_PASS_ENV, "");
+                keyPath = null;
+                certPath = null;
+                return;
+            }    
+        }
```

Comment:
This is much easier to read than it was before, but these anonymous blocks with early returns are still a little bit funky to read. Maybe this is a sign that the class could use some further refactoring to use something like a strategy pattern?

Reply:
I adopted the strategy pattern in the latest version. Please confirm I'm using it correctly. Thanks!